### PR TITLE
remove extra period observed in #15097

### LINKED ISSUE #15097: "Cannot find a physical path" error message has incomplete information when printing
```coq
Require Import Coq.DoesNotExist.
```
The error message seems to be missing some information here:
```
Cannot find a physical path bound to logical path matching suffix Coq..
```

cc @herbelin

### PR DESCRIPTION
**Pull request by Alizter**
I've also refactored some related code, making it more obvious where the period is supposed to appear.

Fixes / closes #15097

- [x] Added / updated **test-suite**.

### CODE REVIEW

File: vernac/vernacentries.ml [hunk 1]
```diff
@@ -464,24 +464,24 @@ let msg_found_library = function
 let err_unmapped_library ?from qid =
   let dir = fst (repr_qualid qid) in
   let prefix = match from with
-  | None -> str "."
+  | None -> str ""
```
Comment (SkySkimmer) [inline vernac/vernacentries.ml:467]: Empty string should be `mt()`

Comment (Alizter) [reply vernac/vernacentries.ml:467]: Thanks! I was wondering about that.

File: test-suite/output/bug_15097.out [hunk 2]
```diff
@@ -0,0 +1,2 @@
+The command has indeed failed with message:
+Cannot find a physical path bound to logical path matching suffix Coq.
```
Comment (SkySkimmer) [inline test-suite/output/bug_15097.out:2]: This error is very strange tbh, the problem is DoesNotExist but all it mentions is Coq.
Why don't we just say `cannot find Coq.DoesNotExist` (`cannot find DoesNotExist with prefix Coq` if using From)?

File: vernac/vernacentries.ml [hunk 3]
```diff
@@ -468,27 +468,37 @@ let err_unmapped_library ?from qid =
...
+end
 
 let print_located_library qid =
   let open Loadpath in
   match locate_qualified_library qid with
   | Ok lib -> msg_found_library lib
-  | Error LibUnmappedDir -> err_unmapped_library qid
-  | Error LibNotFound -> err_notfound_library qid
+  | Error LibUnmappedDir -> raise (UnmappedLibrary (None, qid))
+  | Error LibNotFound    -> raise (NotFoundLibrary (None, qid))
```
Comment (SkySkimmer) [inline vernac/vernacentries.ml:501]:
```suggestion
  | Error LibNotFound -> raise (NotFoundLibrary (None, qid))
```

Comment (Alizter) [reply vernac/vernacentries.ml:500]: bad style?

File: test-suite/output/bug_15097.out (hunk 2 above)
Comment (Alizter) [reply test-suite/output/bug_15097.out:2]: I've reworded the message and changed the information now.

**Comment by Alizter** [comment]
I've made these algebraic exceptions. Will squash those commits soon.

**Review by ejgallego — APPROVED** [review]
LGTM thanks!

**Comment by ejgallego** [comment]
@coqbot: merge now

**Pull request merged**